### PR TITLE
Feature/scat 4788 dashboard status

### PR DESCRIPTION
**Pull request by SrinivasaDontiboyina**
### Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

### CODE REVIEW

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/utils/TendersAPIModelUtils.java [hunk 1]
```diff
@@ -21,8 +23,19 @@
  */
 @Component
 @RequiredArgsConstructor
+@Slf4j
 public class TendersAPIModelUtils {
 
+  private static final List<String> EVALUATING_STATUS_LIST = new ArrayList<>(List.of("Qualification Evaluation","Technical Evaluation",
```
Comment (tommyb82) [inline src/main/java/uk/gov/crowncommercial/dts/scale/cat/utils/TendersAPIModelUtils.java:29]: Remove unnecessary `new ArrayList<>` in these declarations

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementEventService.java [hunk 2]
```diff
@@ -737,20 +735,23 @@ public List<EventSummary> getEventsForProject(final Integer projectId, final Str
...
       if (event.getExternalEventId() == null) {
         var assessment = assessmentService.getAssessment(event.getAssessmentId(), Optional.empty());
         statusCode = TenderStatus.fromValue(assessment.getStatus().toString().toLowerCase());
       } else {
-        var exportRfxResponse = jaggaerService.getRfx(event.getExternalEventId());
-        statusCode = jaggaerAPIConfig.getRfxStatusToTenderStatus()
+         var exportRfxResponse = jaggaerService.getRfx(event.getExternalEventId());
+         statusCode = jaggaerAPIConfig.getRfxStatusToTenderStatus()
             .get(exportRfxResponse.getRfxSetting().getStatusCode());
+         rfxSetting=exportRfxResponse.getRfxSetting();
```
Comment (tommyb82) [inline src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementEventService.java:746]: Formatter settings might be a bit off - we use GoogleStyle, generally speaking (should be plugin for Eclipse & IntelliJ - Venki has details for the latter 👍

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/utils/TendersAPIModelUtils.java [hunk 3]
```diff
@@ -55,6 +68,8 @@ public EventSummary buildEventSummary(final String eventId, final String name,
     eventSummary.setEventStage(stage);
     eventSummary.setStatus(status);
     eventSummary.setEventType(type);
+    //eventSummary.setDashboardStatus();
```
Comment (tommyb82) [inline src/main/java/uk/gov/crowncommercial/dts/scale/cat/utils/TendersAPIModelUtils.java:71]: Remove commented line

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/utils/TendersAPIModelUtils.java [hunk 4]
```diff
@@ -108,6 +125,63 @@ public EventDetail buildEventDetail(final RfxSetting rfxSetting,
...
+                                            final ProcurementEvent procurementEvent) {
+
+    var tenderStatus = procurementEvent.getTenderStatus();
+    if (Objects.nonNull(tenderStatus)) {
+      if (Objects.equals(CLOSED_STATUS, tenderStatus) || Objects.equals(COMPLETE_STATUS, tenderStatus)) {
+        if (Objects.equals(COMPLETE_STATUS, tenderStatus)) {
+          return DashboardStatus.COMPLETE;
+        } else if (Objects.equals(CLOSED_STATUS, tenderStatus)) {
+          return DashboardStatus.CLOSED;
+        }
```
Comment (tommyb82) [inline src/main/java/uk/gov/crowncommercial/dts/scale/cat/utils/TendersAPIModelUtils.java:142]: I think we could remove the outer `if()` here, seems redundant?

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/utils/TendersAPIModelUtils.java [hunk 5]
```diff
@@ -108,6 +125,63 @@ public EventDetail buildEventDetail(final RfxSetting rfxSetting,
...
+      } else  // No rfx, use procurement event status
+        if (Constants.TENDER_DB_ONLY_EVENT_TYPES
+                .contains(DefineEventType.fromValue(procurementEvent.getEventType()))) {
+          return DashboardStatus.ASSESSMENT;
+        } else // TODO: Event types: EOI,RFI,FC OR DA etc
+          if (Constants.TENDER_NON_DB_EVENT_TYPES
+                  .contains(DefineEventType.fromValue(procurementEvent.getEventType()))) {
+            if (Objects.nonNull(rfxSetting)) {
```
Comment (tommyb82) [inline src/main/java/uk/gov/crowncommercial/dts/scale/cat/utils/TendersAPIModelUtils.java:150]: Merge with outer `if(..)`

**Review by tommyb82 — APPROVED** [review]
Great work! And especially with all the test cases.. 🤯 👍 
Only a few minor formatting / syntax comments.

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/utils/TendersAPIModelUtils.java (hunk 4 above)
Comment (SrinivasaDontiboyina) [reply src/main/java/uk/gov/crowncommercial/dts/scale/cat/utils/TendersAPIModelUtils.java:142]: agree. Removed

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/utils/TendersAPIModelUtils.java (hunk 3 above)
Comment (SrinivasaDontiboyina) [reply src/main/java/uk/gov/crowncommercial/dts/scale/cat/utils/TendersAPIModelUtils.java:71]: done

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/utils/TendersAPIModelUtils.java [hunk 6]
```diff
@@ -1,12 +1,14 @@
 package uk.gov.crowncommercial.dts.scale.cat.utils;
 
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.List;
-import java.util.Optional;
+import java.time.Instant;
```
[the remark this answers is not in the document]
Comment (SrinivasaDontiboyina) [reply src/main/java/uk/gov/crowncommercial/dts/scale/cat/utils/TendersAPIModelUtils.java:3]: removed

**Pull request merged**